importlib_resources/_common.py

```diff
@@ -5,21 +5,23 @@
 import contextlib
 import types
 import importlib
+import inspect
 
 from typing import Union, Optional
 from .abc import ResourceReader, Traversable
 
 from ._compat import wrap_spec
 
-Package = Union[types.ModuleType, str]
+Package = Union[types.ModuleType, str, None]
 
 
-def files(package):
+def files(package=None):
     # type: (Package) -> Traversable
     """
     Get a Traversable resource from a package
     """
-    return from_package(resolve(package))
+    context = inspect.currentframe().f_back.f_globals  # type: ignore
```

Comment:
I don't like that this potentially expensive call now gets called unconditionally. Instead, it would be better to resolve it only if no `package` was supplied. That would be more straightforward to do in `resolve` except then, one has an additional level of stack frame (or more) to consider.

Reply:
Additionally, this call relies on `currentframe`, which is [a CPython implementation detail](https://docs.python.org/3/library/inspect.html#inspect.currentframe). It may not be possible to infer the module name.

Reply:
These issues were addressed in commits leading up to 87785ce9dc.